src/led.c

```diff
@@ -75,31 +77,37 @@ static uint32_t led_set_sequence_step(led_data_t *leds, uint32_t step_num)
 
 void led_run_sequence(led_data_t *leds, const led_seq_step_t *sequence, int32_t num_repeat)
 {
-	leds->last_mode = leds->mode;
 	leds->mode = led_mode_sequence;
 	leds->sequence = sequence;
 	leds->seq_num_repeat = num_repeat;
 	led_set_sequence_step(leds, 0);
 	led_update(leds);
 }
 
-void led_indicate_trx(led_data_t *leds, led_num_t num)
-{
-	uint32_t now = HAL_GetTick();
-	led_state_t *led = &leds->led_state[num];
+void led_indicate_trx(led_data_t *leds, led_num_t num) {
+	leds->led_state[num].blink_request = 1;
+}
 
-	if ( SEQ_ISPASSED(now, led->on_until) &&
-		 SEQ_ISPASSED(now, led->off_until) ) {
-		led->off_until = now + 30;
-		led->on_until = now + 45;
+static void led_trx_blinker(led_state_t *ledstate, uint32_t now) {
+	if (!ledstate->blink_request) {
+		return;
 	}
 
-	led_update(leds);
+	ledstate->blink_request = 0;
+
+	if ( SEQ_ISPASSED(now, ledstate->on_until) &&
+		 SEQ_ISPASSED(now, ledstate->off_until) ) {
+		ledstate->off_until = now + 30;
+		ledstate->on_until = now + 45;
+	}
 }
 
 static void led_update_normal_mode(led_state_t *led)
 {
 	uint32_t now = HAL_GetTick();
+
+	led_trx_blinker(led, now);
```

Comment:
nitpick: I would move the check for `led->blink_request` from the callee to the caller:

```suggestion
	if (led->blink_request) {
		led_trx_blinker(led, now);
	}
```

Reply:
That's something I sometimes hesitate about - hiding a simple run/return condition like this inside the func, or have it at the caller level. Doing it from the caller can become heavy if there are multiple call sites (or if the caller shouldn't know about the details of the conditions), but on the other hand it's more obvious that calling the function will do something and shouldn't always be called.

Here it,s all static funcs so both approaches would probably compile to the same code, so I don't feel strongly either way. What is your reasoning for this type of use case ?

Reply:
IMHO it's clearer to read, the function is called `led_update_normal_mode()` so blinking is a special case of normal mode. We only call `blinker` if some condition (`led->blink_request`) is set.

If you are not coming from the blinker use case, but "turn off the LED", then I'm wandering why is `blinker` called unconditionally.

Reply:
Fair. Will change.